### PR TITLE
Documentation on test friendness

### PR DESCRIPTION
**Pull request by Ulmo-F**
Some clues on "how to test private code"

### CODE REVIEW

File: docs/GTest-friend.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+There already are many good resources which explains why you shouldn't test private parts of your implementation and how to avoid it.
+
+If you want to make make your classes friend with their test, here are some ways to achieve it:
```
Comment (TheCoconutChef) [inline docs/GTest-friend.md:3]: Typo `make make`.

Perhaps favor `there are some ways to achieve it`?

File: docs/GTest-friend.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    template <typename ...> struct GTEST;
+    namespace testing::detail { template <char ...> struct string; }
+    
+    class SomeClass {
+        friend struct GTEST<SomeClass, testing::detail::string<> >;
+    };
+
+    GTEST(Game) { ... }
+    ```
+* You can also declare friend with the test class `GTEST("SomeTest")`, it is more cumbersome :
```
Comment (TheCoconutChef) [inline docs/GTest-friend.md:21]: Perhaps favor `although it is more cumbersome` or `though it is more cumbersome`?

**Review by TheCoconutChef — COMMENTED** [review]
Sorry for the drive by review.

**Comment by Ulmo-F** [comment]
Thanks for your comments.
(My english mastery is still arguable)

**Pull request merged**